sound/soc/sof/topology.c

```diff
@@ -2754,7 +2755,10 @@ static int sof_set_dai_config_multi(struct snd_sof_dev *sdev, u32 size,
 				return -ENOMEM;
 
 			/* set cpu_dai_name */
-			dai->cpu_dai_name = link->cpus->dai_name;
+			if (link->num_cpus == 1)
+				dai->cpu_dai_name = link->cpus->dai_name;
+			else
+				dai->cpu_dai_name = link->cpus[j++].dai_name;
```

Comment:
only "dai->cpu_dai_name = link->cpus[j++].dai_name;"  can work ?

Reply:
ha thats what I had but that will increment j for HDA DAIs and it wont work.

Reply:
Should we test `if (j < link->num_cpus)`?

Reply:
yes fixed!

Reply:
so does this mean that if `.num_cpus > 1` then the number of DAI widgets must not exceed it, but if `.num_cpus == 1` then we can have an arbitrary number of DAI widgets?